packages/gestalt/src/Button.js

```diff
@@ -142,7 +157,89 @@ const ButtonWithForwardRef: React$AbstractComponent<
     </Text>
   );
 
+  const iconEndComponent = (): Node => {
+    const SIZE_NAME_TO_PIXEL = {
+      sm: 10,
+      md: 12,
+      lg: 12,
+    };
+
+    return (
+      <Box alignItems="center" display="flex">
+        {buttonText}
+        <Box display="inlineBlock" flex="none" marginStart={2}>
+          <Icon
+            accessibilityLabel=""
+            color={textColor}
+            icon={iconEnd}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
   /* eslint-disable react/button-has-type */
+  if (props.type === 'link') {
```

Comment:
@AlbertCarreras instead of `props.type`, can we compare with `type` and destructure above?

Reply:
can't, it triggers Flow errors